upload/views/helpers.py

```diff
@@ -45,3 +50,45 @@ def get_repository_from_string(
         service=service, repo_identifier=repo_identifier
     )
     return repository
+
+
+def get_repository_and_owner_from_slug_and_commit(
+    slug: str, commitid: str
+) -> tuple[Repository | None, Owner | None]:
+    if "::::" not in slug:
+        return None, None
+
```

Comment:
My understanding for this is to try to find the Repo and Owner based on the commit SHA because there's no service available. But we are ok with an astronomically low probability that we have duplicate entries in the DB with ("username", "repo name", "commit SHA in the repo")?

Reply:
this is a great suggestion - but it would only work if it's guaranteed that the commit already exists in our db. 

With BA and TA, in the views we are doing a `get_or_create` with the incoming commitid, so I took that to mean that the incoming request might be the first upload for that commit. 
commitid is required so the incoming request will always have a commitid, however, if the commit isn't in our db yet then `AND`ing the 3 fields together in this way guarantees no match.

Reply:
er, sorry, my comment above is more suited to your other suggestion.

The possibility exists for us to not be able to find a distinct match. This is a major drawback to this implementation, it means we can't guarantee tokenless uploads on BA and TA for everyone - it depends on what's in our db.

I don't like this, but without `service` or other distinct identifiers for the owner or repo, it's all we can do. I'm hoping that it works for like 95% of uploads, and if it didn't, they could re-upload once the commit is in the db. That's why I'm logging the different outcomes, so I can see how successful it is.